pkg/stdlib/collections/lib.go

```diff
@@ -2,9 +2,9 @@ package collections
 
 import "github.com/MontFerret/ferret/pkg/runtime/core"
 
-func NewLib() map[string]core.Function {
-	return map[string]core.Function{
+func NewLib(ns core.Namespace) error {
+	return ns.RegisterFunctions(core.Functions{
```

Comment:
Looks weird that libraries takes namespaces and register themselves. 

In my opinion, would be better to pass libraries into namespaces.

Reply:
The idea behind this decision is that a library can create nested namespaces within a given one. Plus, in some scenarios you might want to use a specific namespace for a specific library.

Reply:
What about returning the new namespace?